classes/class-truelayer-config-editor.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Config Editor Class file.
+ *
+ * @package TrueLayer_For_WooCommerce/Classes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * The TrueLayer Config Editor class
+ */
+class TrueLayer_Config_Editor {
+	private $config_path;
+	private $backup_path;
+
+	/**
+	 * Class constructor to setup the class.
+	 *
+	 * @param string $path The path to the file you want to edit. Defaults to wp-config.php.
+	 * @param string $backup_path The path to the backup file. Defaults to wp-config-backup.php.
+	 */
+	public function __construct( $path = '', $backup_path = '' ) {
+		$this->config_path = empty( $path ) ? ABSPATH . 'wp-config.php' : $path;
+		$this->backup_path = empty( $backup_path ) ? ABSPATH . 'wp-config-backup.php' : $backup_path;
+	}
+
+	/**
+	 * Add a specific key with a value to the wp-config.php file as a definition.
+	 *
+	 * @param string $key_name The name of the key to add.
+	 * @param string $key_value The value of the key to add.
+	 *
+	 * @return bool True if the key was added, false if it was not.
+	 */
+	public function add_key( $key_name, $key_value ) {
+		$key_name  = $this->sanitize_key_name( $key_name );
+		$key_value = $this->sanitize_key_value( $key_value );
```

Comment:
I can remove it if we want, or at least change it to validate the key to ensure non-safe characters cant be added like for example `'` and rather fail the process. I did actually have a different version of the `sanitize_key_value` method, but somewhere along the line i must have accidentally changed it to match the regex in the key sanitation. I will look into that.

But would you prefer if we don't sanitize and just validate it as a safe string @lighe?

Reply:
I think we should maybe use the original value and simply escape `'` (is there any other character that may cause an issue? I don't think so 🤔

Reply:
Yes i agree, i originally made it with the idea in mind that it could potentially be reused in other places if needed. But for now i think you are correct with handling just the `'` character. However i chose to strip it instead of escaping it, since it should not be a part of the key we generate either way. If you prefer i escape it i can do that instead 🙂